tests/test_data.py

```diff
@@ -94,33 +94,37 @@ def test_empty(self):
         """Check empty patterns match anything"""
         self.assertMatch({})
 
-        self.assertMatch({}, sources=None)
+        self.assertMatch({}, sources=data.Target.ANY)
+        self.assertMatch({}, sources=data.Target.ALL)
+        self.assertMatch({}, sources=data.Target.NONE)
         self.assertMatch({}, sources={"a"})
 
-        self.assertMatch({}, trees=None)
+        self.assertMatch({}, trees=data.Target.ALL)
         self.assertMatch({}, trees={"a"})
 
-        self.assertMatch({}, arches=None)
+        self.assertMatch({}, arches=data.Target.ALL)
         self.assertMatch({}, arches={"a"})
 
-        self.assertMatch({}, sources=None, trees=None)
-        self.assertMatch({}, sources=None, trees={"a"})
-        self.assertMatch({}, sources={"a"}, trees=None)
+        self.assertMatch({}, sources=data.Target.ANY, trees=data.Target.ANY)
+        self.assertMatch({}, sources=data.Target.NONE, trees=data.Target.NONE)
+        self.assertMatch({}, sources=data.Target.ALL, trees=data.Target.ALL)
+        self.assertMatch({}, sources=data.Target.ALL, trees={"a"})
+        self.assertMatch({}, sources={"a"}, trees=data.Target.ALL)
         self.assertMatch({}, sources={"a"}, trees={"a"})
 
     def test_specific_sources(self):
         """Check patterns match specific/all sources correctly"""
         self.assertMismatch({"not": dict(sources=None)},
```

Comment:
Shouldn't these `None` be turned into `data.Target.ANY`?

Reply:
These are not target sets, but pattern nodes, and one of the valid values for them is `None`.  So, no.